libvast/vast/value_index.hpp

```diff
@@ -237,7 +237,7 @@ class arithmetic_index : public value_index {
       } else {
         auto str = caf::get<caf::config_value::string>(i->second);
         auto b = to<base>(str);
-        VAST_ASSERT(b != nullptr); // pre-condition is that this was validated
+        VAST_ASSERT(b.engaged()); // pre-condition is that this was validated
```

Comment:
```suggestion
        VAST_ASSERT(b); // pre-condition is that this was validated
```

You could also rely on the boolean conversion operator. This is perhaps more idiomatic.